### ISSUE TITLE
JSONSchema draft 2019-09 support

### ISSUE THREAD

**Issue by srounce** [microsoft/vscode#98724]
JSONSchema draft-07 is currently supported but `http://json-schema.org/draft/2019-09/schema#` does not currently load, or at least no completions or validation is provided.

**Comment by aeschli** [comment]
The implementation will be in https://github.com/microsoft/vscode-json-languageservice

**Comment by Relequestual** [comment]
As an FYI, OpenAPI Specification 3.1 defines using JSON Schema 2020-12.
We're working on the test suite.
There are a few implementations of 2020-12 already should anyone wish to try it.

**Comment by pratnala** [comment]
I still see this error in VS Code

`Draft 2019-09 schemas are not yet fully supported.`

Is this fixed?

**Comment by ajcool2k** [comment]
> I still see this error in VS Code
> 
> 
> 
> `Draft 2019-09 schemas are not yet fully supported.`
> 
> 
> 
> Is this fixed?

What do you mean by fixed? There is no implementation for the 2019-09 version.

**Comment by Relequestual** [comment]
@ajcool2k what do you mean by "There is no implementation for the 2019-09 version."?
Within VSCode, no there is not.
In the community? There are many: https://json-schema.org/implementations.html

**Comment by ajcool2k** [comment]
> @ajcool2k what do you mean by "There is no implementation for the 2019-09 version."?
> 
> Within VSCode, no there is not.
> 
> In the community? There are many: https://json-schema.org/implementations.html

This is clear. My point is, that there cannot be a fix when there is no bug and clearly there is no bug.

**Comment by Relequestual** [comment]
🤷‍♂️ "Is it fixed?" / "Is my problem resolved now?"
Please try to consider that English is not everyones first language, and be kind. It's quite clear what they meant.

**Comment by ajcool2k** [comment]
> 🤷‍♂️ "Is it fixed?" / "Is my problem resolved now?"
> Please try to consider that English is not everyones first language, and be kind. It's quite clear what they meant.
> 
> > Demonstrating empathy and kindness toward other people
> 
> https://opensource.microsoft.com/codeofconduct/

I apologize, wrote this on the go. Just found it strange that people joining the discussion of an open feature requesting asking if it is fixed yet :-). Yeah but I agree, will be more polite on this in the feature. Thanks for making me aware of it @Relequestual

**Comment by Relequestual** [comment]
Related: https://github.com/microsoft/vscode-json-languageservice/issues/116

**Comment by wolf99** [comment]
Am working on https://github.com/exercism/problem-specifications/issues/1972 and would love to see more recent drafts supported, even small thinhgs like the support for `"format": "uuid"` from 2019-09 would make things instantly happier 🙂.

**Comment by aeschli** [comment]
I made progress and closed some gaps: See https://github.com/microsoft/vscode-json-languageservice/pull/133

The remaining known gaps are:
- formats
- meta-schemas (http://json-schema.org/understanding-json-schema/reference/schema.html)
- bundling (http://json-schema.org/understanding-json-schema/structuring.html#bundling)

**Comment by isc-rsingh** [comment]
Just a cheerleading comment. Looking forward to meta-schema support. I have a project that needs it badly!

**Comment by thomasklein-toptal** [comment]
Cheerleading @aeschli and dancing around my chair for this one to get done :)

**Comment by stephan-probst** [comment]
A question: When using `"$schema": "https://json-schema.org/draft/2020-12/schema` in VS Code 1.71.2 I get the warning `The schema uses meta-schema features ($dynamicRef) that are not yet supported by the validator.`. This is fine and expected. Validation for the implemented features works fine. However, with 2020-12 intellisense / autocompletion in VS code does not work any longer. It works fine if I go to draft-07. Is this a known issue or expected behaviour?

**Comment by aeschli** [comment]
@stephan-probst Can you file an issue with steps and a sample? I quickly tested and I see code completions.

**Comment by gcacars** [comment]
> A question: When using `"$schema": "https://json-schema.org/draft/2020-12/schema` in VS Code 1.71.2 I get the warning `The schema uses meta-schema features ($dynamicRef) that are not yet supported by the validator.`. This is fine and expected. Validation for the implemented features works fine. However, with 2020-12 intellisense / autocompletion in VS code does not work any longer. It works fine if I go to draft-07. Is this a known issue or expected behaviour?

Same here. For root level properties this works fine:
![image](https://user-images.githubusercontent.com/1190940/223701704-b639930c-5cfd-4f45-a050-bf67469873ad.png)

But for sub levels, autocompletion doesn't work:
![image](https://user-images.githubusercontent.com/1190940/223701870-5d3533e8-042c-4f61-bdd2-68a961f67aa5.png)

When I change to `draft-07` the magic happens:
![image](https://user-images.githubusercontent.com/1190940/223703157-3627590a-def0-4edb-a419-0f758f30a900.png)

microsoft/vscode-json-languageservice#179

**Comment by dahu33** [comment]
Is this issue resolved? I'm a bit confused here.

**Comment by gcacars** [comment]
3 years later...

**Comment by Relequestual** [comment]
The VSCode Language Service is now on Bowtie: https://bowtie-json-schema.github.io/bowtie/

Bowtie is a multi-implementation tester that uses the official JSON Schema test suite and reports on compliance. It was created and is maintained by members of the JSON Schema team and community.

**Comment by DaVince** [comment]
> 3 years later...

Some features just take a while to implement, and 3 years later _has_ seen progress, y'know? :)

**Comment by gcacars** [comment]
Yep, but this is most like 2019-19 version, I can imagine that this isn't a big effort.
Also, I just complaining about the auto complete, that I guess that is more simple yet.

**Comment by julian-alarcon** [comment]
OpenAPI 3.0.4 released, and we still have this issue

**Comment by delmohf** [comment]
> OpenAPI 3.0.4 released, and we still have this issue

Even version 3.1 was released in 2021 and this issue is still here :(

**Comment by gcacars** [comment]
5 years later... 🎉😢 Helloooo Microsoft!!!

**Comment by ViliusZur** [comment]
bump

**Comment by BennoDev** [comment]
bumping again

**Comment by keegan-caruso** [comment]
@aeschli I believe I have built support for 2019-09 except remote refs in https://github.com/microsoft/vscode-json-languageservice. How would you like me to organize the contributions?
